tests/stable/check_test.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package stable
+
+import (
+	"flag"
+	"fmt"
+	"testing"
+
+	"log"
+	"os"
+	"strings"
+
+	"github.com/google/go-cmp/cmp"
+	prommodel "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	"gopkg.in/yaml.v3"
+
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+var promText string
+var stableYaml string
+
+var skipStableMetrics = []string{
+}
+
+type Metric struct {
+	Name   string `yaml:"name"`
+	Help   string `yaml:"help"`
+	Type   string
+	Labels []string
+	// Histogram type
+	Buckets []float64 `yaml:"buckets,omitempty"`
+}
+
+func TestMain(m *testing.M) {
+	flag.StringVar(&promText, "collectedMetricsFile", "", "input prometheus metrics text file, text format")
+	flag.StringVar(&stableYaml, "stableMetricsFile", "", "expected stable metrics yaml file, yaml format")
+	flag.Parse()
+	m.Run()
+}
+
+func TestStableMetrics(t *testing.T) {
+	mf, err := parsePromText(promText)
+	fatal(err)
+	collectedStableMetrics := extractStableMetrics(mf)
+	printMetric(collectedStableMetrics)
+
+	expectedStableMetrics, err := readYaml(stableYaml)
+	if err != nil {
+		t.Fatalf("Can't read stable metrics from file. err = %v", err)
+	}
+
+	err = compare(collectedStableMetrics, *expectedStableMetrics, skipStableMetrics)
```

Comment:
Honestly, i would just sort the metrics, output a yaml string and do a simple string comparison. 

This is not only simpler, but when you generate the yaml string, you'll end up with deterministic ordering which will make the diff more pleasant.